autogluon/utils/tabular/ml/models/knn/knn_utils.py

```diff
@@ -0,0 +1,184 @@
+import numpy as np
+from scipy.stats import mode
+from sklearn.utils.extmath import weighted_mode
+import faiss
+
+import logging
+
+logger = logging.getLogger(__name__)
+
+# Rather than try to import non-public sklearn internals, we implement our own weighting functions here
+# These support the same operations as the sklearn functions - at least as far as possible with FAISS
+def _check_weights(weights):
```

Comment:
minor nit comments relating to code readability: If you haven't already I'd recommend adding PEP8 violation highlighting to your IDE. While we don't follow the line-length limit PEP8 recommendation, we do follow most others, such as having 2 new lines between each def and class, as well as 2 new lines after imports before code is written.

This isn't critical, as I can clean-up the code afterwards, but is something to keep in mind in future PR's to make the process smoother.

Reply:
Thanks for the tip, I didn't know you could do this. I did it and formatted everything in a separate commit.